resource/systemd/unit/unit.go

```diff
@@ -0,0 +1,543 @@
+	If "ignore-requirements" it will start a unit but only ignore the requirement
+	dependencies. It is not recommended to make use of the latter two options.
+	Returns the newly created job object.
+	*/
+	StartMode systemd.StartMode // how to start the unit
+
+	// the amount of time the command will wait for configuration to load
+	// before halting forcefully. The
+	// format is Go's duraction string. A duration string is a possibly signed
+	// sequence of decimal numbers, each with optional fraction and a unit
+	// suffix, such as "300ms", "-1.5h" or "2h45m". Valid time units are "ns",
+	// "us" (or "µs"), "ms", "s", "m", "h".
+	Timeout time.Duration // how long to wait for the load state to resolve
+}
+
+// Check if all the properties for the unit are correct
+// 1. Checks if the unit is currently loading, if so waits Default 5 seconds
+// 2. Checks if the unit is in the active state
+// 3. Check if the unit is in the unit file state
+func (t *Unit) Check(r resource.Renderer) (resource.TaskStatus, error) {
+	systemd.ApplyDaemonReload()
+	/*////////////////////////////////////////////////////
+	First thing to do is to check if the Name given is outside
+	the normal search path of systemd. If so it should be linked
+	*/ ////////////////////////////////////////////////////
+	dir, unitName := filepath.Split(t.Name)
+	var shouldBeLinked = false
+	if dir != "" && t.UnitFileState.IsLinked() {
+		// The fullpath to a unit was given. Assume that this isn't in the
+		// normal search paths
+		shouldBeLinked = true
+	}
+
+	// Get the connection from the pool
+	conn, err := systemd.GetDbusConnection()
+	if err != nil {
+		status := &resource.Status{
+			WarningLevel: resource.StatusFatal,
+			Status:       err.Error(),
+		}
+		return status, err
+	}
+	defer conn.Return()
+	dbusConn := conn.Connection
+
+	//Create context
+	var (
+		ctx    context.Context
+		cancel context.CancelFunc
+	)
+	if t.Timeout == 0 {
+		t.Timeout = systemd.DefaultTimeout
+	}
+	ctx, cancel = context.WithTimeout(context.Background(), t.Timeout)
+	defer cancel()
+
+	// Waits for the daemon to finish loading.
+	// After check for unit linking
+	err = systemd.WaitToLoad(ctx, dbusConn, unitName)
+
+	if err != nil {
+		// Unit doesn't exist should be linked
+		if err.Error() == "LoadError: \"[\\\"org.freedesktop.DBus.Error.FileNotFound\\\", \\\"No such file or directory\\\"]\"" {
+			if shouldBeLinked {
+				// Check if file to be linked exist on disk
+				_, statErr := os.Stat(t.Name)
+				if statErr != nil {
+					return &resource.Status{
+						Status:       err.Error(),
+						WarningLevel: resource.StatusFatal,
+						Output:       []string{err.Error()},
+					}, err
+				}
+				statusMsg := fmt.Sprintf("unit %q does not exist, will be linked", unitName)
+				diffMsg := fmt.Sprintf("unit %q does not exist", unitName)
+				diff := resource.TextDiff{Default: diffMsg, Values: [2]string{diffMsg, fmt.Sprintf("unit %q is linked", unitName)}}
+				return &resource.Status{
```

Comment:
status should be embedded in the Task now. Check the module authors' guide at converge.aster.is